app/routes/user/password.tsx

```diff
@@ -0,0 +1,230 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+import type { DataFunctionArgs } from '@remix-run/node'
+import { useFetcher } from '@remix-run/react'
+import {
+  Alert,
+  Button,
+  FormGroup,
+  Icon,
+  InputGroup,
+  Label,
+  TextInput,
+} from '@trussworks/react-uswds'
+import { useEffect, useRef, useState } from 'react'
+import invariant from 'tiny-invariant'
+
+import { getUser } from '../__auth/user.server'
+import { updatePassword } from './password.server'
+import PasswordValidationStep from '~/components/PasswordValidationStep'
+import Spinner from '~/components/Spinner'
+import { getFormDataString } from '~/lib/utils'
+import { useUserIdp } from '~/root'
+
+export const handle = {
+  breadcrumb: 'Reset Password',
+  getSitemapEntries: () => null,
+}
+
+export async function loader({ request }: DataFunctionArgs) {
+  const user = await getUser(request)
+  if (!user) throw new Response(null, { status: 403 })
+  return null
+}
+
+export async function action({ request }: DataFunctionArgs) {
+  const data = await request.formData()
+  const oldPassword = getFormDataString(data, 'oldPassword')
+  const newPassword = getFormDataString(data, 'newPassword')
+
+  invariant(oldPassword && newPassword)
+  return await updatePassword(request, oldPassword, newPassword)
+}
+
+export function WrongIdp() {
+  return (
+    <Alert type="error" heading="Error:" headingLevel="h4">
+      You must be signed in with a username and password to reset your password.
+    </Alert>
+  )
+}
+
+function ErrorMessage(errorMessage: any, isOldPasswordTouched: boolean) {
```

Comment:
Transform this to a React functional component so that we can invoke this using JSX syntax:

```suggestion
function ErrorMessage({errorMessage, isOldPasswordTouched}: {errorMessage: any, isOldPasswordTouched: boolean}) {
```